packages/toolpad-app/src/toolpad/AppEditor/HierarchyExplorer/validation.ts

```diff
@@ -0,0 +1,9 @@
+import * as React from 'react';
+import * as appDom from '../../../appDom';
+
+export function useNodeNameValidation(name: string, disallowedNames: Set<string>, kind: string) {
```

Comment:
If we use that scenario a lot, maybe we could have an option to already check for and exclude the existing names here?

Reply:
I did it this way for now because most places this is used in also use the `existingNames` to propose a default value for the node. And I wanted to avoid calculating this list multiple times. 🤔 maybe I should memoize it like the `getChildNodes` function

Reply:
Yeah that makes sense, we can keep it as it is in that case, wouldn't make sense to change the hook.